holoviews/core/data/grid.py

```diff
@@ -600,14 +615,20 @@ def range(cls, dataset, dimension):
             expanded = cls.irregular(dataset, dimension)
             column = cls.coords(dataset, dimension, expanded=expanded, edges=True)
         else:
-            column = dataset.dimension_values(dimension)
+            column = cls.values(dataset, dimension, flat=False)
         if column.dtype.kind == 'M':
-            return column.min(), column.max()
+            dmin, dmax = column.min(), column.max()
+            if da and isinstance(column, da.Array):
+                return dmin.compute(), dmax.compute()
```

Comment:
Would be better to do `da.compute(dmin, dmax)`. That reuses any common intermediate results for both computations. Also only optimizes the underlying Dask graph once.